jax/lax/lax.py

```diff
@@ -2809,6 +2809,7 @@ def _scatter_add_transpose_rule(t, operand, scatter_indices, updates,
     operand_t = t
 
   if updates is None:
+    t = ad.instantiate_zeros(operand, t)
```

Comment:
Hmm is it possible for `operand` to be None here? I think yes:

```python
grad(lambda x: lax.tie_in(jax.ops.index_add(x, [0], x)[0], 7.))(np.ones(1))
```

I think we might need to add `operand_shape` to the params of `scatter_add`, since otherwise we don't have enough information to instantiate this zero.

That'd be in keeping with the way we've solved this kind of issue in the past, at least. There might be other options.

Reply:
Come to think of it, why do we need to instantiate zeros here instead of just propagating them? I did that instead.